frontend-react/src/components/USLink.tsx

```diff
@@ -1,38 +1,116 @@
-import React, { PropsWithChildren } from "react";
+import React, { AnchorHTMLAttributes } from "react";
 import { Link, NavLink } from "react-router-dom";
 
 // Known issue with the `PropsWithChildren generic in React 18,
 // so I wrote this in a way that we can just remove `<any>` and be
 // okay if we update our React version.
-interface USNavLinkProps extends PropsWithChildren<any> {
-    href: string;
+interface CustomLinkProps {
+    children: React.ReactNode | React.ReactNode[];
+    anchor?: boolean;
     className?: string;
     activeClassName?: string;
 }
-type USLinkProps = Omit<USNavLinkProps, "activeClassName">;
+type USLinkProps = AnchorHTMLAttributes<{}> &
+    Omit<CustomLinkProps, "activeClassName">;
+type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
-/** A single link for rendering standard links */
-export const USLink = ({ children, href, className }: USLinkProps) => (
-    <Link to={href} className={`usa-link ${className}`}>
-        {children}
-    </Link>
-);
+/** Helper class to protect against the string literal "undefined" appearing in styles
```

Comment:
Do we need this?  We have the `classnames` package already, so we could just use that instead:

```js
const className1 = undefined;
const className2 = null;
const className3 = "style-c";
classnames("style-a style-b", className1, className2, className3); // returns "style-a style-b style-c"
```

Reply:
I had no idea we even had that! Saves me time and trouble (or, I guess, _would have_ saved me) 😂 I'll update this